### PR TITLE
3body

### PR DESCRIPTION
**Pull request by schererc**
Additions to the votca package in order to parametrize 3-body interactions of the SW type with csg_fmatch and calculate additional properties with csg_stat

For votca/csg#274

### CODE REVIEW

File: include/votca/tools/cubicspline.h [hunk 1]
```diff
@@ -99,6 +99,20 @@ class CubicSpline : public Spline
     void AddToFitMatrix(matrix_type &A, double x,
             int offset1, int offset2=0, double scale=1);
 
+    /**
+     * \brief Add a point (one entry) to fitting matrix
+     * \param pointer to matrix
+     * \param value x
+     * \param offsets relative to getInterval(x)
+     * \param scale1 parameters for terms "A,B,C,D"
```
Comment (JoshuaSBrown) [inline include/votca/tools/cubicspline.h]: Maybe add [in] [out] brackets next to the param keyword to indicate if the param is being changed?

**Comment by junghans** [comment]
@schererc, have a look at https://github.com/votca/tools/blob/master/src/tests/test_matrix.cc for an example unit test.

**Comment by junghans** [comment]
Ping @schererc

File: include/votca/tools/cubicspline.h (hunk 1 above)
Comment (schererc) [reply include/votca/tools/cubicspline.h]: I now have added the brackets

**Comment by schererc** [comment]
I now have added a unit test for cubicspline.h in test/test_cubicspline.cc

File: src/tests/test_cubicspline.cc [hunk 2]
```diff
@@ -72,6 +72,32 @@ BOOST_AUTO_TEST_CASE(cubicspline_fit_test) {
   
 }
 
+BOOST_AUTO_TEST_CASE(cubicspline_matrix_test) {
+    
+  CubicSpline cspline;   
+  cspline.setBCInt(0);
+  cspline.GenerateGrid(0.4,0.6,0.1);
+  Eigen::MatrixXd A=Eigen::VectorXd::Zero(1,6);
```
Comment (JensWehner) [inline src/tests/test_cubicspline.cc]: this should be matrices and not vectors

File: src/tests/test_cubicspline.cc [hunk 3]
```diff
@@ -72,6 +72,32 @@ BOOST_AUTO_TEST_CASE(cubicspline_fit_test) {
...
+  Eigen::MatrixXd Aref=Eigen::VectorXd::Zero(1,6);
```
Comment (JensWehner) [inline src/tests/test_cubicspline.cc]: same here

File: src/tests/test_cubicspline.cc (hunk 2 above)
Comment (schererc) [reply src/tests/test_cubicspline.cc]: Sure!

File: src/tests/test_cubicspline.cc (hunk 3 above)
Comment (schererc) [reply src/tests/test_cubicspline.cc]: Sure!

File: src/tests/test_cubicspline.cc (hunk 2 above)
Comment (schererc) [reply src/tests/test_cubicspline.cc]: I also found and corrected another error in test_cubicspline.cc. Now it should work

File: include/votca/tools/cubicspline.h [hunk 4]
```diff
@@ -99,6 +99,20 @@ class CubicSpline : public Spline
     void AddToFitMatrix(matrix_type &A, double x,
             int offset1, int offset2=0, double scale=1);
 
+    /**
+     * \brief Add a point (one entry) to fitting matrix
+     * \param pointer to matrix [in] [out]
```
Comment (JensWehner) [inline include/votca/tools/cubicspline.h]: reference to matrix not pointer, but I am not sure how consistent the rest is.

File: src/tests/test_cubicspline.cc [hunk 5]
```diff
@@ -72,6 +72,32 @@ BOOST_AUTO_TEST_CASE(cubicspline_fit_test) {
...
 
+BOOST_AUTO_TEST_CASE(cubicspline_matrix_test) {
+    
+  CubicSpline cspline;   
+  cspline.setBCInt(0);
+  cspline.GenerateGrid(0.4,0.6,0.1);
+  Eigen::MatrixXd A=Eigen::MatrixXd::Zero(1,6);
+  Eigen::MatrixXd Aref=Eigen::MatrixXd::Zero(1,6);
+  
+  Aref(0,0)=0.0;
```
Comment (JensWehner) [inline src/tests/test_cubicspline.cc]: instead of this you can simply write `Aref<<0.0,-9.0,10.0,0.0,-0.3333333,-0.0166667;`

but it is fine this way too.

Comment (schererc) [reply src/tests/test_cubicspline.cc]: Thanks

File: include/votca/tools/cubicspline.h (hunk 4 above)
Comment (schererc) [reply include/votca/tools/cubicspline.h]: The rest is also not more consistent

Comment (JensWehner) [reply include/votca/tools/cubicspline.h]: Yeah I know, but it is still wrong @junghans

Comment (junghans) [reply include/votca/tools/cubicspline.h]: ?

Comment (junghans) [reply include/votca/tools/cubicspline.h]: Well, make at least the next comments correct!

**Comment by junghans** [comment]
Waiting for @JoshuaSBrown then we can merge!

**Comment by JensWehner** [comment]
I'd say we should merge now.

**Pull request merged**